vyper/builtin_functions/functions.py

```diff
@@ -1980,8 +2098,7 @@ def fetch_call_return(self, node):
         # figure out the output type by converting
         # the types to ABI_Types and calling size_bound API
         arg_abi_types = []
-        for arg in node.args:
-            arg_t = get_exact_type_from_node(arg)
+        for arg_t in self.infer_arg_types(node):
```

Comment:
nit: don't iterate over a call which might have side effects. best practice would be to have the side effects line separate from the loop line:
```suggestion
        arg_types = self.infer_arg_types(node)
        for arg_t in arg_types:
```